tests/unit/src/main/scala/tests/BuildServerInitializer.scala

```diff
@@ -0,0 +1,135 @@
+import org.eclipse.lsp4j.MessageActionItem
+
+sealed trait BuildServerInitializer {
+  def initialize(
+      workspace: AbsolutePath,
+      server: TestingServer,
+      client: TestingClient,
+      layout: String,
+      expectError: Boolean
+  )(implicit ec: ExecutionContext): Future[Unit]
+}
+
+/**
+ * Set up your workspace using QuickBuild as your build tool.
+ * This will take your `metals.json` file and quickly produce `.bloop/` files from it.
+ */
+object QuickBuildInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      workspace: AbsolutePath,
+      server: TestingServer,
+      client: TestingClient,
+      layout: String,
+      expectError: Boolean
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+    QuickBuild.bloopInstall(workspace)
+    for {
+      _ <- server.initialize()
+      _ <- server.initialized()
+    } yield {
+      if (!expectError) {
+        server.assertBuildServerConnection()
+      }
+    }
+  }
+}
+
+/**
+ * Set up your workspace by responding to an Import Build request which will
+ * run Bloop Install via the build tool being used.
+ */
+object BloopImportInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      workspace: AbsolutePath,
+      server: TestingServer,
+      client: TestingClient,
+      layout: String,
+      expectError: Boolean
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+    for {
+      _ <- server.initialize()
+      // Import build using Bloop
+      _ = client.importBuild = ImportBuild.yes
+      _ <- server.initialized()
+    } yield {
+      if (!expectError) {
+        server.assertBuildServerConnection()
+      }
+    }
+  }
+}
+
+/**
+ * Assumes sbt is being used as a build tool and also for your BSP server.
+ * This generates the .bsp/sbt.json file and invoke the BSP switch command
+ * with sbt as the build server.
+ */
+object SbtServerInitializer extends BuildServerInitializer {
+  this: BaseLspSuite =>
+  override def initialize(
+      workspace: AbsolutePath,
+      server: TestingServer,
+      client: TestingClient,
+      layout: String,
+      expectError: Boolean
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+    generateBspConig(workspace)
+    for {
+      _ <- server.initialize()
+      _ <- server.initialized()
+      // choose sbt as the Bsp Server
+      _ = client.selectBspServer = { _ => new MessageActionItem("sbt") }
+      _ <- server.executeCommand(ServerCommands.BspSwitch.id)
+    } yield {
+      if (!expectError) {
+        server.assertBuildServerConnection()
+      }
+    }
+  }
+
+  private def generateBspConig(workspace: AbsolutePath): Unit = {
+    val bspFolder = workspace.resolve(".bsp")
+    val sbtJson = bspFolder.resolve("sbt.json")
+    // don't overwrite existing BSP config
+    if (!sbtJson.isFile) {
+      // we create bsp/sbt.json file manually because `sbt bspConfig` takes too long
+      val sbtLaunchJar =
```

Comment:
Are you able to just use `SbtBuildTool` and run it?

Reply:
That's tested in `SbtServerSuite.generate`. I am not using `SbtBuildTool` here to save some time by not running `sbt bspConfig`.